src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -177,37 +180,27 @@
-       copied via RunTests script so VS can work when the test dir is initially clean.
-       -->
+  <!-- This copies the same files as the RunTests script, when running locally.
+       This achieves two things: 
+	   1) Allows VS to work when the test dir is initially clean, since VS does not invoke the runner script
+        because it needs to attach to the XUnit process when debugging.
+	   2) For incremental build of any kind, ensures that any test dependency which is updated gets copied
+       (thanks to smarter copy logic) -->
   <PropertyGroup Condition="'$(BuildingInsideVisualStudio)'=='true'">
-    <PrepareForRunDependsOn>$(PrepareForRunDependsOn);CopyDefaultTestAssetsForVS</PrepareForRunDependsOn>
+    <PrepareForRunDependsOn>$(PrepareForRunDependsOn);CopyDependenciesToTestDirectory</PrepareForRunDependsOn>
   </PropertyGroup>
-  <Target Name="CopyDefaultTestAssetsForVS" DependsOnTargets="CopyTestToTestDirectory;CopySupplementalTestData">
-     <!-- This was copied from RunTestsForProject in tests.targets
-          The RunTestsForProject target does not execute in VS context and would be confused by the script based runner.
-          _TestCopyLocalByFileNameWithoutDuplicates are the precise items that are fed to the runner script generation code.
-     -->
-    <ItemGroup>
-      <!-- Not all platforms can use the .ni.dlls that come from packages.  If TestWithoutNativeImages is specified, we'll exclude them from copy generation.
-           If we end up needing this for any other sorts of filtering, we'll want to add a list of filtered extensions to be matched on EndsWith.  -->
-      <_IncludedFileForTestsInVS Include="@(_TestCopyLocalByFileNameWithoutDuplicates)"
-                                 Condition="'$(TestWithoutNativeImages)' != 'true' Or !$([System.String]::Copy('%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)').EndsWith('.ni.dll'))" >
-        <PackageRelativePath Condition="'%(_TestCopyLocalByFileNameWithoutDuplicates.NugetPackageId)' != ''">$([System.String]::Copy('%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)').Replace('$(PackagesDir)',''))</PackageRelativePath>
-        <UseAbsolutePath Condition="'$(TestWithLocalLibraries)'=='true'">$([System.String]::Copy('%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)').StartsWith('$(BinDir)'))</UseAbsolutePath>
-        <DestinationPath>$(TestPath)\%(Filename)%(Extension)</DestinationPath>
-      </_IncludedFileForTestsInVS>
-      <_IncludedFileForTestsInVs Remove="@(_IncludedFileForTestsInVS)" Condition="Exists('%(DestinationPath)')" />
-    </ItemGroup>
-
+  <Target Name="CopyDependenciesToTestDirectory"
+          Condition="'$(DisableCopyTestToTestDirectory)' != 'true' And '$(ArchiveTests)' != 'true'"
+          DependsOnTargets="CopyTestToTestDirectory;DiscoverTestDependencies">
     <Copy
-      SourceFiles="@(_IncludedFileForTestsInVS  -> '%(SourcePath)')"
```

Comment:
As we discussed offline, the issue is that the code archives the test runtime directory rather than the test project's bin directory, so it needs to copy only the files from the bin directory to the test runtime directory in the case where it will be archived, which required the two Copy operations.

We discussed changing it to archive the test project's bin directory instead.  It looks like the only additional files beyond what's already in that directory that we'd have to add to the archive are:
- The test execution script (RunTest.cmd/sh)
- The project.json and project.lock.json files (currently added at https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets#L227)
- `$(BUILDTOOLS_OVERRIDE_RUNTIME)/*.*`

I'm wondering if `BUILDTOOLS_OVERRIDE_RUNTIME` is even used anymore, though.  #670 says that it hasn't worked since April, and it looks like it is currently used in init-tools.sh in buildtools but not init-tools.cmd. @weshaggard or @ellismg or @joperezr , do you know if it's still supported?  Even if it is still supported, would it ever be used in a Helix run, or can I assume that it won't be used if `ArchiveTests` is true?